### PR TITLE
product api add column

### LINKED ISSUE #16: API Design
I already deployed a online Swagger UI. (From no-57/orange-backstage-api#23)

- [Frontstage](http://146.235.17.81:8081/)
- [Backstage](http://146.235.17.81:8080/)

Please try to implement it according to this Spec for frontstage as much as possible. Thanks.

### PR DESCRIPTION
**Pull request by asdfg429**
## add columns
    1. distributor;
    2. price;
    3. discount;
    4. path;

### CODE REVIEW

File: src/main/java/com/orange/main/price/bo/Price.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    public void setProductId(int productId) {
+        this.productId = productId;
+    }
+
+
+
+    private BigDecimal price;
+    private BigDecimal discount;
+    private String source_url;
```
Comment (shadow3x3x3) [inline src/main/java/com/orange/main/price/bo/Price.java]: IIUC, you should no align the field name to the DB column name. On the contrary, keeping the JAVA naming style.

File: src/main/java/com/orange/main/price/repo/PriceRepository.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.orange.main.price.repo;
+
+
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import com.orange.main.price.bo.Price;
+
+public interface PriceRepository  extends PagingAndSortingRepository<Price, Long>{
```
Comment (shadow3x3x3) [inline src/main/java/com/orange/main/price/repo/PriceRepository.java]: style: extra and missing space

```suggestion
public interface PriceRepository extends PagingAndSortingRepository<Price, Long> { 
```

File: src/main/java/com/orange/main/products/bo/ProductDTO.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package com.orange.main.products.bo;
+
+import java.math.BigDecimal;
+
+
+public class ProductDTO {
+
+    public ProductDTO(Products products, String distributor, BigDecimal price, BigDecimal discount, String path){
```
Comment (shadow3x3x3) [inline src/main/java/com/orange/main/products/bo/ProductDTO.java]: ditto

```suggestion
    public ProductDTO(Products products, String distributor, BigDecimal price, BigDecimal discount, String path) {
```

File: src/main/java/com/orange/main/products/bo/ProductDTO.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        this.products = products;
+        if(null != this.products.getPrices()){
+            this.products.setPrices(null);
+        }
```
Comment (shadow3x3x3) [inline src/main/java/com/orange/main/products/bo/ProductDTO.java]: This logic should not be belonged to this class. Handling it with a method of product instead.

File: src/main/java/com/orange/main/products/service/impl/ProductsServiceImpl.java [hunk 5]
```diff
@@ -32,19 +39,42 @@ public Products getProductById(Long productId) {
...
+            for(Products product:rtn){
+                Price p = product.getPrices()
+                .stream()
+                .min(Comparator.comparing(Price::getPrice))
+                .orElseThrow(NoSuchElementException::new);
+                ProductDTO productDto = new ProductDTO(product, p.getSellerType().toString(), p.getPrice(), p.getDiscount(), null==product.getProductImgs()?null:product.getProductImgs().getPath());
+                dto.add(productDto);
+           }
+        }
+        return dto;
```
Comment (shadow3x3x3) [inline src/main/java/com/orange/main/products/service/impl/ProductsServiceImpl.java]: Using early return instead.

```suggestion
        if (CollectionUtils.isEmpty(rtn)) {
            return dto;
        }
        
        for(Products product : rtn) {
                Price p = product.getPrices()
                .stream()
                .min(Comparator.comparing(Price::getPrice))
                .orElseThrow(NoSuchElementException::new);
                ProductDTO productDto = new ProductDTO(product, p.getSellerType().toString(), p.getPrice(), p.getDiscount(), null==product.getProductImgs()?null:product.getProductImgs().getPath());
                dto.add(productDto);
        }
        return dto;
```

**Review by shadow3x3x3 — COMMENTED** [review]
Please have a look.

**Comment by asdfg429** [comment]
[fix by comments & add endpoint for get images](https://github.com/No-57/orange-frontstage-api/pull/23/commits/f4b81557df730105a0bdb237a88ece3068c4d939)

**Review by shadow3x3x3 — APPROVED** [review]

**Pull request merged**